tee-worker/litentry/core/assertion-build/src/a14.rs

```diff
@@ -114,26 +51,28 @@ pub fn build(
 		}
 	}
 	let mut value = false;
-	let mut client = A14Client::new(data_provider_config);
-
-	for address in polkadot_addresses {
-		let data = A14Data {
-			params: A14DataParams { address },
-			include_metadata: false,
-			include_widgets: false,
-		};
-		let response = client.send_request(&data)?;
-
-		let result = response
-			.data
-			.get("result")
-			.and_then(|r| r.as_bool())
-			.ok_or(Error::RequestVCFailed(Assertion::A14, ErrorDetail::ParseError))?;
-		if result {
-			value = result;
-			break
-		}
-	}
+	let mut client = AchainableClient::new(data_provider_config);
+
+	loop_with_abort_strategy(
+		polkadot_addresses,
+		|address| {
+			let data = LabelQueryReq {
+				params: LabelQueryReqParams { address: address.clone() },
+				include_metadata: false,
+				include_widgets: false,
+			};
+			let result = client
+				.query_label("a719e99c-1f9b-432e-8f1d-cb3de0f14dde", &data)
+				.map_err(|e| Error::RequestVCFailed(Assertion::A14, e.into_error_detail()))?;
+			if result {
+				value = result;
+				Ok(LoopControls::Break)
+			} else {
+				Ok(LoopControls::Continue)
+			}
+		},
+		AbortStrategy::ContinueUntilEnd::<fn(&_) -> bool>,
```

Comment:
So i tracked the logic of this, If we are continuing until end, we are only propagating the last error we received in the loop, Does it make sense to accumulate all the errors and propagate that?  Or maybe if not for the client, but for some kind of observability?

Reply:
Hey @felixfaisal all errors are now being accumulated now in this: https://github.com/litentry/litentry-parachain/pull/2707/commits/bf386dcec959135abb60ac8f0ce0d8217426bdcb